### DISCUSSION TITLE
Two letters locale identifier vs UTS format locale

### DISCUSSION THREAD

**Discussion by petertulala** [vercel/next.js#23662] [Help]
I'm trying to set up locales detection:

```
module.exports = {
  i18n: {
    defaultLocale: 'en',
    locales: ['en', 'de', 'sk', 'cs', 'tr'],
  },
};
```

All works well if the `Accept-Language` header is exactly the `sk` or `de`. But if the user send the `Accept-Language` header in UTS format, for instance `sk-SK`, the locale is not detected as `sk` and the user is not redirected to `/sk` path.

How to configure Next.js in a way that locales `de`, `de-DE`, `de-AT`, `de-CH`, `de-BE`, `de-LU` are all treated as `de`? Similarly, `tr`, `tr-TR`, `tr-CY` should be all treated as `tr`.

I want to have just one path for each of the languages -- i.e. one path for German language, one path for Turkish language, etc. I don't want to use paths like `/de-CH` in my app.

**Comment by DCzajkowski** [comment]
I think this ^ should be the default, but it's not supported at all?

**Comment by Jordankay-brighthr** [comment]
I'm also trying to do the above, how can this be done?
